generic_config_updater/generic_updater.py

```diff
@@ -18,9 +21,94 @@ def release_lock(self):
         pass
 
 class ChangeApplier:
-    def apply(self, change):
-        # TODO: Implement change applier
-        raise NotImplementedError("ChangeApplier.apply(change) is not implemented yet")
+    def __init__(self, service_metadata_file: str=None):
+        self.config_db = ConfigDBConnector()
+        self.config_db.connect()
+        service_metadata = {}
+        if service_metadata_file:
+            with open(service_metadata_file, 'r') as f:
+                service_metadata = json.load(f)
+        self.table_service = service_metadata.get('tables', {})
+        self.service_commands = service_metadata.get('services', {})
+        self.valiate_metadata()
+        self.previous_table_name = None
+
+    def valiate_metadata(self):
+        # TODO: check metadata file syntax
+        pass
+
+    def exec_command(self, cmd) -> int:
+        p = subprocess.Popen(cmd)
+        p.communicate()
+        return p.returncode
+
+    def on_table_operation(self, table_name):
+        services = self.table_service.get(table_name, {})
+        for service in services:
+            commands = self.service_commands.get(service, {})
+            restart_comand = commands.get('restart-command')
+            validate_command = commands.get('validate-commands')
+            if restart_comand:
+                rc = self.exec_command(restart_comand)
+                if rc != 0:
+                    raise GenericConfigUpdaterError(f"Restart command failed: {restart_comand}, rc={rc}")
+                rc = self.exec_command(validate_command)
```

Comment:
Validate commands expect ConfigDb before and after the update, from design doc:
>validate-commands - Multiple CLI commands used to validate the service has absorbed the change successfully. The validate commands can expect the config before and after the update to passed to the CLI commands for detailed validation if needed.

This was requested by linkedin team

Reply:
We discussed this with LinkedIn: Zhenggen Xu <zxu@linkedin.com>; Praveen Chaudhary <pchaudhary@linkedin.com> on 3/15, from the notes:
>
![image](https://user-images.githubusercontent.com/3927651/132536812-cf1cfd15-584f-47e1-8c0b-7404e8550133.png)

Specifically when they delete a port they not only verify it was deleted from ConfigDb, but also from AsicDB. Check code [here](https://github.com/Azure/sonic-utilities/blob/2b12aadeed3a86ca4ede3aa30b451914c4acb00b/config/config_mgmt.py#L380).

So in order to support their request, we are passing to the validation function what commands were deleted/added. I just made it a little bit generic and passed the whole configDb before and after the update. The validate script itself can do diffing to identify these commands that were changed, or they can loop over the PORTs and check which was deleted.